src/main/java/htsjdk/samtools/seekablestream/ByteArraySeekableStream.java

```diff
@@ -27,7 +27,11 @@ public long position() throws IOException {
 
     @Override
     public void seek(long position) throws IOException {
-        this.position = position;
+        if (position < 0) {
+			throw new IOException("Negative seek offset");
```

Comment:
This should probably be an `IllegalArgumentException`, and I would make the exception include the requested position as well.

```java
throw new IllegalArgumentException("Cannot seek to a negative position, position=" + position + ".");
```